File: tests/benchmark/corpus/php/safe/safe_cross_function_dotdot.php
<?php
// php-safe-016: cross-function bool-returning validator with rejection.

function validate_no_dotdot(string $s): bool {
    return strpos($s, '..') === false && $s[0] !== '/' && $s[0] !== '\\';
}

function handle(string $userPath): void {
    if (!validate_no_dotdot($userPath)) {
        return;
    }
    file_get_contents($userPath);
}
?>

File: tests/benchmark/corpus/java/safe/SafeCrossFunctionDotdot.java
// java-safe-016: cross-function bool-returning validator with rejection.
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class SafeCrossFunctionDotdot {
    public static boolean validateNoDotdot(String s) {
        return !s.contains("..") && !s.startsWith("/") && !s.startsWith("\\");
    }

    public static void handle(String userPath) throws IOException {
        if (!validateNoDotdot(userPath)) {
            return;
        }
        Files.readAllBytes(Paths.get(userPath));
    }
}
